src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -174,14 +175,7 @@ and FSharpProjectOptions =
             && options1.UnresolvedReferences = options2.UnresolvedReferences
             && options1.OriginalLoadReferences = options2.OriginalLoadReferences
             && options1.ReferencedProjects.Length = options2.ReferencedProjects.Length
-            && (options1.ReferencedProjects, options2.ReferencedProjects)
-               ||> Array.forall2 (fun r1 r2 ->
```

Comment:
So all this is redundant? Do you have an idea why was it there in the first place?

Reply:
The referenced projects union is a relatively new addition to FCS, previously it could only be additional `FSharpProjectOptions` instances. When it was initially added, it only contained `PEReference` and `FSharpReference` cases, so I think it was natural to modify the check by adding the second branch next to existing options one. This is the source of the bug: the third case was added later and this check was overlooked.

Reply:
Alright, thanks!